doc: Fix various typos

dimitris-t opened the pull request:
"on a already existing" -> "on an already existing". Change made on two occurrences.

dimitris-t committed:
Fixed multiple typos

A few "a->an" and "an->a".
"Shows, if the supplied default SOCKS5 proxy" -> "Shows if the supplied default SOCKS5 proxy". Change made on 3 occurrences.
"without fully understanding the ramification of a command" -> "without fully understanding the ramifications of a command".
Removed duplicate words such as "the the".

practicalswift commented:
ACK 7540cdb94fdc54387cf1bd5b345af8e7c9ee881e

paveljanik commented:
Please unify all three typo PRs into this one.

jonasschnelli reviewed (approved):
ACK 7540cdb94fdc54387cf1bd5b345af8e7c9ee881e

jonasschnelli commented:
Can you combine the typo PRs?
Add 8964c1c83360c188561607e2a03af44e354f8d60 and aeabc7a8e1fc0f6ed1c0be9f7266a5b63717de31 here and change the PR description?
Ideally, typo-/trivial-PRs should be kept at a minimum because they take away precious review time.
However, thank you for fixing the typos.

dimitris-t commented:
I hope I did this correctly. I fixed the other typos in the same dimitris-t:patch-1 branch as this PR. Updated this PR's description. Closed the other two PRs.

practicalswift commented:
ACK 83d81483d6a15a7615aeace73003c283a24ee9e5

paveljanik commented:
Can you fix all a -> an? E.g.
```
src/rpc/net.cpp:                            "4. \"absolute\"     (boolean, optional) If set, the bantime must be a absolute timestamp in seconds since epoch (Jan 1 1970 GMT)\n"
```

paveljanik commented:
Few more:

```
src/qt/bitcoinstrings.cpp:"Error loading %s: You can't enable HD on a already existing non-HD wallet"),
src/qt/bitcoinstrings.cpp:QT_TRANSLATE_NOOP("bitcoin-core", "Error loading %s: You can't disable HD on a already existing HD wallet"),
src/qt/clientmodel.cpp:        //pass a async signal to the UI thread
test/functional/rest.py:        # GETUTXOS: now query a already spent outpoint #
```

And squash afterwards, please.

dimitris-t commented:
A comment inside of bitcoinstrings.cpp says it is auto-generated, should I touch it?

practicalswift commented:
When fixing a-vs-an typos, please consider fixing these as well:

```diff
diff --git a/contrib/rpm/bitcoin.if b/contrib/rpm/bitcoin.if
index 2b096c2..b206866 100644
--- a/contrib/rpm/bitcoin.if
+++ b/contrib/rpm/bitcoin.if
@@ -121,7 +121,7 @@ interface(`bitcoin_manage_lib_dirs',`
 ########################################
 ## <summary>
 ##     All of the rules required to administrate
-##     an bitcoin environment
+##     a bitcoin environment
 ## </summary>
 ## <param name="domain">
 ##     <summary>
diff --git a/src/httpserver.h b/src/httpserver.h
index 6be9950..9df56e5 100644
--- a/src/httpserver.h
+++ b/src/httpserver.h
@@ -86,7 +86,7 @@ public:

     /**
      * Get the request header specified by hdr, or an empty string.
-     * Return an pair (isPresent,string).
+     * Return a pair (isPresent,string).
      */
     std::pair<bool, std::string> GetHeader(const std::string& hdr);

@@ -125,7 +125,7 @@ public:
     virtual ~HTTPClosure() {}
 };

-/** Event class. This can be used either as an cross-thread trigger or as a timer.
+/** Event class. This can be used either as a cross-thread trigger or as a timer.
  */
 class HTTPEvent
 {
diff --git a/src/rest.cpp b/src/rest.cpp
index 8fb0c13..33e3fb4 100644
--- a/src/rest.cpp
+++ b/src/rest.cpp
@@ -413,7 +413,7 @@ static bool rest_getutxos(HTTPRequest* req, const std::string& strURIPart)
         boost::split(uriParts, strUriParams, boost::is_any_of("/"));
     }

-    // throw exception in case of a empty request
+    // throw exception in case of an empty request
     std::string strRequestMutable = req->ReadBody();
     if (strRequestMutable.length() == 0 && uriParts.size() == 0)
         return RESTERR(req, HTTP_BAD_REQUEST, "Error: empty request");
diff --git a/src/test/data/script_tests.json b/src/test/data/script_tests.json
index e35a7ce..0390d68 100644
--- a/src/test/data/script_tests.json
+++ b/src/test/data/script_tests.json
@@ -2506,7 +2506,7 @@
 ],

 ["CHECKSEQUENCEVERIFY tests"],
-["", "CHECKSEQUENCEVERIFY", "CHECKSEQUENCEVERIFY", "INVALID_STACK_OPERATION", "CSV automatically fails on a empty stack"],
+["", "CHECKSEQUENCEVERIFY", "CHECKSEQUENCEVERIFY", "INVALID_STACK_OPERATION", "CSV automatically fails on an empty stack"],
 ["-1", "CHECKSEQUENCEVERIFY", "CHECKSEQUENCEVERIFY", "NEGATIVE_LOCKTIME", "CSV automatically fails if stack top is negative"],
 ["0x0100", "CHECKSEQUENCEVERIFY", "CHECKSEQUENCEVERIFY,MINIMALDATA", "UNKNOWN_ERROR", "CSV fails if stack top is not minimally encoded"],
 ["0", "CHECKSEQUENCEVERIFY", "CHECKSEQUENCEVERIFY", "UNSATISFIED_LOCKTIME", "CSV fails if stack top bit 1 << 31 is set and the tx version < 2"],
diff --git a/src/validation.cpp b/src/validation.cpp
index bef1733..7b40447 100644
--- a/src/validation.cpp
+++ b/src/validation.cpp
@@ -1206,7 +1206,7 @@ static bool CheckInputs(const CTransaction& tx, CValidationState &state, const C
                             return state.Invalid(false, REJECT_NONSTANDARD, strprintf("non-mandatory-script-verify-flag (%s)", ScriptErrorString(check.GetScriptError())));
                     }
                     // Failures of other flags indicate a transaction that is
-                    // invalid in new blocks, e.g. a invalid P2SH. We DoS ban
+                    // invalid in new blocks, e.g. an invalid P2SH. We DoS ban
                     // such nodes as they are not following the protocol. That
                     // said during an upgrade careful thought should be taken
                     // as to the correct behavior - we may want to continue
diff --git a/test/functional/rest.py b/test/functional/rest.py
index fbcceba..6e58090 100755
--- a/test/functional/rest.py
+++ b/test/functional/rest.py
@@ -82,9 +82,9 @@ class RESTTest (BitcoinTestFramework):
                 n = vout['n']


-        ######################################
-        # GETUTXOS: query a unspent outpoint #
-        ######################################
+        #######################################
+        # GETUTXOS: query an unspent outpoint #
+        #######################################
         json_request = '/checkmempool/'+txid+'-'+str(n)
         json_string = http_get_call(url.hostname, url.port, '/rest/getutxos'+json_request+self.FORMAT_SEPARATOR+'json')
         json_obj = json.loads(json_string)
@@ -97,9 +97,9 @@ class RESTTest (BitcoinTestFramework):
         assert_equal(json_obj['utxos'][0]['value'], 0.1)


-        ################################################
-        # GETUTXOS: now query a already spent outpoint #
-        ################################################
+        #################################################
+        # GETUTXOS: now query an already spent outpoint #
+        #################################################
         json_request = '/checkmempool/'+vintx+'-0'
         json_string = http_get_call(url.hostname, url.port, '/rest/getutxos'+json_request+self.FORMAT_SEPARATOR+'json')
         json_obj = json.loads(json_string)
@@ -171,14 +171,14 @@ class RESTTest (BitcoinTestFramework):
         #do some invalid requests
         json_request = '{"checkmempool'
         response = http_post_call(url.hostname, url.port, '/rest/getutxos'+self.FORMAT_SEPARATOR+'json', json_request, True)
-        assert_equal(response.status, 400) #must be a 400 because we send a invalid json request
+        assert_equal(response.status, 400) #must be a 400 because we send an invalid json request

         json_request = '{"checkmempool'
         response = http_post_call(url.hostname, url.port, '/rest/getutxos'+self.FORMAT_SEPARATOR+'bin', json_request, True)
-        assert_equal(response.status, 400) #must be a 400 because we send a invalid bin request
+        assert_equal(response.status, 400) #must be a 400 because we send an invalid bin request

         response = http_post_call(url.hostname, url.port, '/rest/getutxos/checkmempool'+self.FORMAT_SEPARATOR+'bin', '', True)
-        assert_equal(response.status, 400) #must be a 400 because we send a invalid bin request
+        assert_equal(response.status, 400) #must be a 400 because we send an invalid bin request

         #test limits
         json_request = '/checkmempool/'
```

practicalswift commented on src/rpc/net.cpp:
```diff
@@ -499,10 +499,10 @@ UniValue setban(const JSONRPCRequest& request)
                             "setban \"subnet\" \"add|remove\" (bantime) (absolute)\n"
                             "\nAttempts add or remove a IP/Subnet from the banned list.\n"
```
Nit: `s/remove a IP/remove an IP/`

practicalswift replied:
Nit: `s/Attempts add or remove/Attempts to add or remove/`

practicalswift commented on src/rpc/net.cpp:
```diff
@@ -499,10 +499,10 @@ UniValue setban(const JSONRPCRequest& request)
                             "\nArguments:\n"
-                            "1. \"subnet\"       (string, required) The IP/Subnet (see getpeerinfo for nodes ip) with a optional netmask (default is /32 = single ip)\n"
+                            "1. \"subnet\"       (string, required) The IP/Subnet (see getpeerinfo for nodes ip) with an optional netmask (default is /32 = single ip)\n"
                             "2. \"command\"      (string, required) 'add' to add a IP/Subnet to the list, 'remove' to remove a IP/Subnet from the list\n"
```
Nit: `s/remove a IP/remove an IP/`

practicalswift commented on src/rpc/net.cpp:
```diff
@@ -499,10 +499,10 @@ UniValue setban(const JSONRPCRequest& request)
                             "\nAttempts add or remove a IP/Subnet from the banned list.\n"
                             "\nArguments:\n"
-                            "1. \"subnet\"       (string, required) The IP/Subnet (see getpeerinfo for nodes ip) with a optional netmask (default is /32 = single ip)\n"
+                            "1. \"subnet\"       (string, required) The IP/Subnet (see getpeerinfo for nodes ip) with an optional netmask (default is /32 = single ip)\n"
```
Nit: `s/ip/IP/g`

practicalswift commented on src/rpc/net.cpp:
```diff
@@ -499,10 +499,10 @@ UniValue setban(const JSONRPCRequest& request)
-                            "1. \"subnet\"       (string, required) The IP/Subnet (see getpeerinfo for nodes ip) with a optional netmask (default is /32 = single ip)\n"
+                            "1. \"subnet\"       (string, required) The IP/Subnet (see getpeerinfo for nodes ip) with an optional netmask (default is /32 = single ip)\n"
                             "2. \"command\"      (string, required) 'add' to add a IP/Subnet to the list, 'remove' to remove a IP/Subnet from the list\n"
                             "3. \"bantime\"      (numeric, optional) time in seconds how long (or until when if [absolute] is set) the ip is banned (0 or empty means using the default time of 24h which can also be overwritten by the -bantime startup argument)\n"
```
Nit: `s/ip/IP/g`

dimitris-t commented:
If we are also considering "a->an" in comments, there is a ton of them that show up with:

`grep -i -E -R --exclude-dir locale "\ba [eioa]" *`

I will do all of those then. I am performing those changes on the GitHub GUI. Can I squash my commits on the GUI or is this something you will do when you merge the PR?

MarcoFalke commented:
The preferred way to squash your commits is mentioned in https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits
Please note that the GitHub GUI does not support squashing, you'd need to do it locally on your machine.

practicalswift commented:
@dimitris-t Converted into `git grep` and excluding false positives with `grep`:
```
$ git grep -E "\ba [eioa]" | egrep -v '(/(locale|release-notes|secp256k1|univalue)/|\.a|\.xpm|%a|for a in)'
src/chain.cpp:    // as it's too large for a arith_uint256. However, as 2**256 is at least as large
src/cuckoocache.h:    /** epoch_heuristic_counter is used to determine when a epoch might be aged
src/cuckoocache.h:     * epoch. When the number of non-erased elements in a epoch
src/qt/bitcoinstrings.cpp:"Error loading %s: You can't enable HD on a already existing non-HD wallet"),
src/qt/bitcoinstrings.cpp:QT_TRANSLATE_NOOP("bitcoin-core", "Error loading %s: You can't disable HD on a already existing HD wallet"),
src/qt/clientmodel.cpp:        //pass a async signal to the UI thread
src/qt/rpcconsole.cpp:    //don't allow a insane font size
src/rpc/net.cpp:                            "1. \"subnet\"       (string, required) The IP/Subnet (see getpeerinfo for nodes ip) with a optional netmask (default is /32 = single ip)\n"
src/rpc/net.cpp:                            "4. \"absolute\"     (boolean, optional) If set, the bantime must be a absolute timestamp in seconds since epoch (Jan 1 1970 GMT)\n"
src/test/arith_uint256_tests.cpp:// element of Aarray and Barray, and then converting the result into a arith_uint256.
src/torcontrol.cpp: * and maintain a ephemeral hidden service.
src/utilstrencodings.cpp:    // Note that strtol returns a *long int*, so even if strtol doesn't report a over/underflow
src/utilstrencodings.cpp:    // Note that strtoll returns a *long long int*, so even if strtol doesn't report a over/underflow
src/utilstrencodings.cpp:    // Note that strtoul returns a *unsigned long int*, so even if it doesn't report a over/underflow
src/utilstrencodings.cpp:    // Note that strtoull returns a *unsigned long long int*, so even if it doesn't report a over/underflow
src/wallet/wallet.cpp:            InitError(strprintf(_("Error loading %s: You can't disable HD on a already existing HD wallet"), walletFile));
src/wallet/wallet.cpp:            InitError(strprintf(_("Error loading %s: You can't enable HD on a already existing non-HD wallet"), walletFile));
src/wallet/wallet.cpp:        //this might happen if a user uses a old wallet within a pruned node
test/functional/rest.py:        assert_equal(len(json_obj['utxos']), 0) #there should be a outpoint because it has just added to the mempool
test/functional/rest.py:        assert_equal(len(json_obj['utxos']), 1) #there should be a outpoint because it has just added to the mempool
test/functional/zapwallettxes.py:        #there must be a expection because the unconfirmed wallettx0 must be gone by now
```

dimitris-t commented:
ok, I figured out how to squash the commits using the command line, I will get on with the typo fixes and squash once more at the end.

dimitris-t commented:
I believe I have done all the requested changes, except for src/qt/bitcoinstrings.cpp that has a comment saying it is auto-generated. Squashed everything into 1 commit.

practicalswift commented:
utACK 4b4f6469adf24234e686b8c9d8ba3bc739266ce3

Good work!

sipa commented:
utACK 4b4f6469adf24234e686b8c9d8ba3bc739266ce3

paveljanik commented:
utACK 4b4f646

dimitris-t commented:
With `grep --colour -i -E -R --exclude-dir pixmaps "\b([a-zA-Z]+)\s+\1\b" *` you can spot a number of words repeated by mistake, such as:

> Stating whether matching outputs should be be treated as not incoming payments

I can see about 12 such occurrences. Shall I tackle this here or leave it for a new PR?

fanquake commented:
@dimitris-t Might as well include it here.

practicalswift commented:
@dimitris-t Please include them here. Very good cleanup work! Keep it coming

dimitris-t commented:
Removed duplicate words, ready for review.

practicalswift commented:
utACK b2665227548b1a614f8bf699c3c0b7ece3d38796

laanwj commented:
utACK on the changes;

nit: Can you please add a newline in your commit message between the subject line and the description?
Otherwise, tooling (such as git shortlog) will put all the entire message on one line:
```
* b266522 Fixed multiple typos A few "a->an" and "an->a". "Shows, if the supplied default SOCKS5 proxy" -> "Shows if the supplied default SOCKS5 proxy". Change made on 3 occurrences. "without fully understanding the ramification of a command" -> "without fully understanding the ramifications of a command". Removed duplicate words such as "the the". (Dimitris Tsapakidis) (pull/10633/head)
```

dimitris-t commented:
There should be a newline after "Fixed multiple typos" now.

practicalswift commented:
utACK 0a5a6b90bc75a8d75fdff71c9b08abe9b0b96b0a

flack commented:
@dimitris-t Seems your regex to find duplicate words didn't take line breaks into account, e.g. https://github.com/bitcoin/bitcoin/search?utf8=%E2%9C%93&q=%22the+the%22&type=

Just in case you feel like doing a followup :-)

dimitris-t commented:
Thanks, I will definitely do a follow-up, now that I got the hang of it.

MarcoFalke commented:
Could you submit the changes to subtrees upstream, please.

I.e. https://github.com/bitcoin-core/secp256k1 for the secp256k1 changes.

dimitris-t commented:
Sure, the other upstream one I see is https://github.com/bitcoin-core/leveldb. If there is anything else please let me know.

MarcoFalke commented:
The others are listed in this section: https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#git-subtree-checksh

dimitris-t commented:
Here they are:
https://github.com/bitcoin-core/secp256k1/pull/462
https://github.com/bitcoin-core/leveldb/pull/6

practicalswift commented:
@dimitris-t Some additional typos that you could tackle in a follow up PR:

```
# a vs an ("an already", "an one-off")
src/qt/bitcoinstrings.cpp:"Error loading %s: You can't enable HD on a already existing non-HD wallet"),
src/qt/bitcoinstrings.cpp:QT_TRANSLATE_NOOP("bitcoin-core", "Error loading %s: You can't disable HD on a already existing HD wallet"),
src/test/netbase_tests.cpp:    // and that a one-off resolves correctly

# duplicate words ("command")
doc/Doxyfile.in:# popen()) the command command input-file, where command is the value of the

# typo ("transfer")
src/httpserver.cpp:    // tranfer ownership to eventBase/HTTP via .release()
```
